raiden/ui/cli.py

```diff
@@ -392,55 +394,62 @@ def run(ctx, **kwargs):
         # - Ask for confirmation to quit if there are any locked transfers that did
         # not timeout.
         (listen_host, listen_port) = split_endpoint(kwargs['listen_address'])
-        with socket_factory(listen_host, listen_port) as mapped_socket:
-            kwargs['mapped_socket'] = mapped_socket
-
-            app_ = ctx.invoke(app, **kwargs)
-
-            domain_list = []
-            if kwargs['rpccorsdomain']:
-                if ',' in kwargs['rpccorsdomain']:
-                    for domain in kwargs['rpccorsdomain'].split(','):
-                        domain_list.append(str(domain))
-                else:
-                    domain_list.append(str(kwargs['rpccorsdomain']))
-
-            if ctx.params['rpc']:
-                raiden_api = RaidenAPI(app_.raiden)
-                rest_api = RestAPI(raiden_api)
-                api_server = APIServer(
-                    rest_api,
-                    cors_domain_list=domain_list,
-                    web_ui=ctx.params['web_ui'],
-                )
-                (api_host, api_port) = split_endpoint(kwargs["api_address"])
-                api_server.start(api_host, api_port)
+        try:
+            with socket_factory(listen_host, listen_port) as mapped_socket:
```

Comment:
This is not correct. The socket factory is used as a context manager because to clean the socket, so the mapped_socket will be closed after the try-except block and unavailable to the application.